services/galley/src/Galley/API/Teams.hs

```diff
@@ -111,12 +111,17 @@ updateTeam (zusr::: zcon ::: tid ::: req ::: _) = do
     push1 $ newPush1 zusr (TeamEvent e) r & pushConn .~ Just zcon
     pure empty
 
-deleteTeam :: UserId ::: ConnId ::: TeamId ::: JSON -> Galley Response
-deleteTeam (zusr::: zcon ::: tid ::: _) = do
-    alive <- Data.isTeamAlive tid
-    membs <- Data.teamMembers tid
-    when alive $
-        void $ permissionCheck zusr DeleteTeam membs
+deleteTeam :: UserId ::: ConnId ::: TeamId ::: Request ::: Maybe JSON ::: JSON -> Galley Response
+deleteTeam (zusr::: zcon ::: tid ::: req ::: _ ::: _) = do
+    team <- Data.team tid
+    case team of
+        Just t | Data.tdDeleted t == False -> do
```

Comment:
The only case that we are interested in (i.e, the team is _alive_) is when the team is a `Just` and the `Deleted` property is `False`.
In every other case, we just want to continue and delete the team.